smoke-tests/apps/LiveMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/LiveMetricsTest.java

```diff
@@ -29,12 +41,68 @@ abstract class LiveMetricsTest {
 
   @Test
   @TargetUri("/test")
-  void doMostBasicTest() throws Exception {
-    testing.getTelemetry(0);
-
+  void testTelemetryDataFlow() throws java.lang.Exception {
     Awaitility.await()
-        .atMost(Duration.ofSeconds(10))
-        .until(() -> testing.mockedIngestion.isLiveMetricsPingReceived());
+        .atMost(Duration.ofSeconds(20))
+        .until(() -> testing.mockedIngestion.getCountForType("RequestData") == 1);
+
+    assertThat(testing.mockedIngestion.isPingReceived()).isTrue();
+
+    List<String> postBodies = testing.mockedIngestion.getPostBodies();
+    assertThat(postBodies).hasSizeGreaterThan(0); // should post at least once
+
+    boolean foundExceptionDoc = false;
+    boolean foundTraceDoc = false;
+    boolean foundDependency = false;
+    boolean foundRequest = false;
+
+    for (String postBody : postBodies) {
+      // Each post body is a list with a singular MonitoringDataPoint
+      List<MonitoringDataPoint> dataPoints = new ArrayList<>();
+      try {
+        JsonReader reader = JsonProviders.createReader(postBody);
+        // See live metrics swagger to understand MonitoringDataPoint structure
+        dataPoints = reader.readArray(MonitoringDataPoint::fromJson);
+      } catch (IOException e) {
+        throw new java.lang.Exception("Failed to parse post request body", e);
+      }
+
+      // Because the mock ping/posts should succeed, we should only have one MonitoringDataPoint per
+      // post
+      assertThat(dataPoints).hasSize(1);
+      MonitoringDataPoint dataPoint = dataPoints.get(0);
+      List<DocumentIngress> docs = dataPoint.getDocuments();
+      List<MetricPoint> metrics = dataPoint.getMetrics();
+
+      // check that the expected documents are present
+      // With the default filtering configuration, we should only see the exception and trace
+      // documents.
+      for (DocumentIngress doc : docs) {
+        if (doc.getDocumentType().equals(DocumentType.EXCEPTION)
+            && ((Exception) doc).getExceptionMessage().equals("Fake Exception")) {
+          foundExceptionDoc = true;
+        } else if (doc.getDocumentType().equals(DocumentType.TRACE)
+            && ((Trace) doc).getMessage().equals("This message should generate a trace")) {
+          foundTraceDoc = true;
+        }
+      }
+
+      // See if dependency and request are counted
+      for (MetricPoint metric : metrics) {
+        String name = metric.getName();
+        double value = metric.getValue();
+        if (name.equals("\\ApplicationInsights\\Dependency Calls/Sec") && value == 1) {
+          foundDependency = true;
+        } else if (name.equals("\\ApplicationInsights\\Requests/Sec") && value == 1) {
+          foundRequest = true;
+        }
+      }
+    }
+
+    assertThat(foundExceptionDoc).isTrue();
+    assertThat(foundTraceDoc).isTrue();
+    assertThat(foundDependency).isTrue();
+    assertThat(foundRequest).isTrue();
```

Comment:
It may be faster/easier to read: 

 ```suggestion
    assertThat(hasException(docs).isTrue();
    assertThat(hasTrace(docs).isTrue();
    assertThat(hasDependency(metrics)).isTrue();
    assertThat(hasRequest(metrics).isTrue();
```

Reply:
Agree that this needs to be more readable, but will do this in a different way than suggested